### ISSUE TITLE
apps/lwaftr/nh_fwd.lua:126: variable 'ipv4_header' is not declared

### ISSUE THREAD

**Issue by mwiget** [Igalia/snabbswitch#520]
upgraded snabb to commit 33d941b 'Add SnabbVMX manual' and run into an issue when using the vMX for next hop resolution. The problem disappears when using static next_hop_mac.

```
root@e320879d0671:/tmp# cat snabbvmx-lwaftr-xe0.cfg
return {
  lwaftr = "snabbvmx-lwaftr-xe0.conf",
  settings = {
  },
  ipv6_interface = {
    ipv6_address = "",
    cache_refresh_interval = 1,
  },
  ipv4_interface = {
    ipv4_address = "172.20.1.16",
    cache_refresh_interval = 1,
  },
}
root@e320879d0671:/tmp# cat snabbvmx-lwaftr-xe0.conf
hairpinning = false,
binding_table = binding_table_60k.txt.s,
aftr_ipv6_ip = 2001:db8::1,
aftr_ipv4_ip = 172.20.1.16,
aftr_mac_inet_side = 02:b0:53:89:03:00,
inet_mac = 02:02:02:02:02:02,
aftr_mac_b4_side = 02:b0:53:89:03:00,
next_hop6_mac = 02:02:02:02:02:02,
ipv4_mtu = 9000,
ipv6_mtu = 9000,
vlan_tagging = false,
icmpv6_rate_limiter_n_packets = 5,
icmpv6_rate_limiter_n_seconds = 1,
```

```
mwiget@lwaftr5> vhost_user: Read cached features (0x10028001) from /tmp/vhost_features_xe0.socket
Set features 0x10028001
 VIRTIO_NET_F_CTRL_VQ VIRTIO_NET_F_MRG_RXBUF VIRTIO_RING_F_INDIRECT_DESC VIRTIO_NET_F_CSUM
rxavail = 0 rxused = 0
rxavail = 23 rxused = 23
apps/lwaftr/nh_fwd.lua:126: variable 'ipv4_header' is not declared
stack traceback:
    core/main.lua:137: in function <core/main.lua:135>
    [C]: in function 'error'
    lib/lua/strict.lua:37: in function '__index'
    apps/lwaftr/nh_fwd.lua:126: in function 'send_ipv4_cache_trigger'
    apps/lwaftr/nh_fwd.lua:227: in function 'method'
    core/app.lua:87: in function 'with_restart'
    core/app.lua:335: in function 'thunk'
    core/histogram.lua:98: in function 'breathe'
    core/app.lua:273: in function 'main'
    program/snabbvmx/lwaftr/lwaftr.lua:196: in function 'run'
    program/snabbvmx/snabbvmx.lua:17: in function 'run'
    core/main.lua:56: in function <core/main.lua:43>
    [C]: in function 'xpcall'
    core/main.lua:179: in main chunk
    [C]: at 0x00452360
    [C]: in function 'pcall'
    core/startup.lua:3: in main chunk
    [C]: in function 'require'
    [string "require "core.startup""]:1: in main chunk
```

**Comment by dpino** [comment]
I realized about that a few days ago and pushed a PR (#513). The problem was that CI tests are failing now. The SnabbVMX selftests are failing when run inside a container.

We have run the tests manually in #513 and it's merged now. The issue should be fixed now.

**Comment by dpino** [comment]
I wait for your confirmation before closing the issue.

**Comment by mwiget** [comment]
@dpino, thanks for the explaining and subsequent fix. I confirm this issue to be resolved when I upgraded to commit bd3811f.
